### DISCUSSION TITLE
Default Kafka (aiokafka) consumer group

### DISCUSSION THREAD

**Discussion by SirSilver** [ag2ai/faststream#1867] [Q&A] [answered]
Hello
I didn't find any provided way to specify default consumer group (`group_id`) for a `KafkaBroker` or `KafkaRouter` for `aiokafka` library on initialization or somewhere else besides `subscriber` decorator. I would like to specify consumer group for a whole set of routes within router or broker from some configuration. I do understand that I can use module level instance of `pydantic_settings` to configure every `subscriber` decorator, but is there any other way?
Or maybe this use case is something not natural for this library?

**Comment by Lancetnik** [comment, marked as answer]
@SirSilver sorry, we have no ability to set consumer group to all broker / routers
But you can add it by yourself easely: just add required params to broker & router `__init__` and then set subscribers' consumer group as `group or self.group` - https://github.com/airtai/faststream/blob/main/faststream/kafka/broker/registrator.py#L1602

**Reply by SirSilver** [reply]
Ok, thank you
